### ISSUE TITLE
g2core via Motate on an LPC CPU (Smoothieboard)

### ISSUE THREAD

**Issue by xpix** [synthetos/Motate#8]
Hello,

if it possible to build motate for an LPC1769? 
It's the cpu of the smoothieboard and it would be great to use this as CNC Controller.

I found this here for GRBL:
https://github.com/gnea/grbl-LPC

**Comment by giseburt** [comment]
Absolutely! Anyone willing to give it a go we'll support the effort as much as we can. Pull requests are always welcome!

As for the technical side, it's completely practical. It will just take some time and effort.

-Rob

**Comment by RadicalZephyr** [comment]
Hi there, I'm really interested in a port to LPC as well (so I can run g2core!), and would love some pointers on how to get started.  I'm fairly new to the world of embedded development so this might be an overly ambitious project for me, but I'm very interested in learning.

In terms of the layering of Motate, my understanding is that the architecture level is handled because the LPC1769 is an ARM Cortex M3, but that the platform, processor and board would support would all need to be created for this.

So I have a few questions on how to get started.

From my understanding, all of these files live inside the `MotateProject/motate` directory, with the processor layer being folders in that directory, and then arch, platform and board related things under the subdirectories of the corresponding names. Is that correct?

Second, which "end" should I start work from?  It seems most natural to start at the lowest level that needs to be implemented (platform) and work my way up, but I think the knowledge contained in the board level is a little more obvious to me.

Third, would it be okay to open a WIP/tracking PR that we can use to focus discussion and feedback with around code changes needed for this?

Finally, do you have any pointers on specific learning resources for this kind of work? :smile: 

Thanks!
